lib/LedgerSMB/Admin/Command/destroy.pm

```diff
@@ -0,0 +1,110 @@
+
+package LedgerSMB::Admin::Command::destroy;
+
+=head1 NAME
+
+LedgerSMB::Admin::Command::destroy - ledgersmb-admin 'destroy' command
+
+=cut
+
+use strict;
+use warnings;
+
+use LedgerSMB::Admin::Command;
+use LedgerSMB::Database;
+
+use Moose;
+extends 'LedgerSMB::Admin::Command';
+use namespace::autoclean;
+
+sub run {
+    my ($self, $dbname, $newname) = @_;
+    my $logger = $self->logger;
+    my $existing_db = $self->connect_data_from_arg($dbname);
+    my $connect_data = {
+        $self->config->get('connect_data')->%*,
+        $existing_db->%*,
+    };
+    my $db = LedgerSMB::Database->new(
+        connect_data => $connect_data,
+        );
+    my $dbh = $db->connect;
+    $dbh->do(q{SELECT setting__set('role_prefix',
+                             coalesce((setting_get('role_prefix')).value, ?))},
+                 {},
+                 "lsmb_$existing_db->{dbname}__")
+        or die $dbh->errstr;
+    $dbh->commit;
+    $dbh->disconnect;
+
+    $db->drop
+        or die $db->errstr;
+    $logger->info('Database successfully destroyed');
+
+    my $connect_admin = {
+        $self->config->get('connect_data')->%*,
+        $existing_db->%*,
+        dbname => ($self->config->get('admindb') // 'postgres'),
+    };
+    $self->db(
+        LedgerSMB::Database->new(
+            connect_data => $connect_admin,
+        ));
+    $dbh = $self->db->connect;
+    my $sth = $dbh->prepare("SELECT rolname FROM pg_roles WHERE rolname LIKE 'lsmb_$existing_db->{dbname}__%';");
```

Comment:
But what if the database had `role_prefix` set? In that case, the roles you're trying to query for here, don't exist. (Mind you: it's not guaranteed that the roles are shared with another database, even when role_prefix is set...)

Reply:
Rewritten to delete only if roles are not used anymore.